pkg/client/unversioned/request.go

```diff
@@ -115,7 +115,7 @@ type Request struct {
 }
 
 // NewRequest creates a new request helper object for accessing runtime.Objects on a server.
-func NewRequest(client HTTPClient, verb string, baseURL *url.URL, groupVersion unversioned.GroupVersion, codec runtime.Codec, backoff BackoffManager) *Request {
+func NewRequest(client HTTPClient, verb string, baseURL *url.URL, versionedAPIPath string, groupVersion unversioned.GroupVersion, codec runtime.Codec, backoff BackoffManager) *Request {
```

Comment:
If versionedAPIPath is supposed to overlap with any prefix in baseURL, please a) don't do that, or b) document document document, and I prefer a) unless there's a super good reason?